src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy

```diff
@@ -95,7 +95,7 @@ abstract class WorkflowUsingMergedBams extends Workflow {
             // Check the cache for the dataset. If not, try to load the files.
             if (!foundBamFilesForDatasets.containsKey(dataSet)) {
 
-                runtimeService.getAllBamFiles(context).collect { BasicBamFile bam ->
+                runtimeService.metadataAccessor.getAllBamFiles(context).collect { BasicBamFile bam ->
```

Comment:
Yes, I see -- although files need not be accessed here. Getting the metadata from actual files by accessing them in the filesystem (well their names) is only one option. The file names could also just get generated from the metadata table using the filename patterns. Then it would still rather be a metadata access.

Maybe the structure is something like MetadataAccessor and FilenameGenerator (e.g. from metadata or actual filesystem accesses), which may or may not communicate with each other in one or the other direction (metadata -> filename vs. filename -> parsed metadata). This is at least the functionality that is implemented in the moment.

Note that the current refactoring is just a first step as an attempt to sort things out. Let's discuss about how we could improve further.